source/common/network/socket_option_factory.cc

```diff
@@ -98,12 +98,24 @@ std::unique_ptr<Socket::Options> SocketOptionFactory::buildLiteralOptions(
                 socket_option.DebugString());
       continue;
     }
+
+    absl::optional<Network::Socket::Type> socket_type = absl::nullopt;
+    if (socket_option.has_type() && socket_option.type().has_stream()) {
+      if (socket_option.has_type() && socket_option.type().has_datagram()) {
```

Comment:
nit: `socket_option.has_type()` was already validated in the line above.

Reply:
Done.